src/bwamem.cpp

```
@@ -982,30 +1014,55 @@ void mem_chain_seeds(FMI_search *fmi, const mem_opt_t *opt,
+                    {
+                        int blo = 0, bhi = chains_n;
+                        while (blo < bhi) {
+                            int mid = (blo + bhi) >> 1;
+                            if (chains_arr[mid].pos < tmp.pos) blo = mid + 1;
+                            else bhi = mid;
+                        }
+                        ins_lo = (blo < chains_n && chains_arr[blo].pos == tmp.pos) ? blo + 1 : blo;
+                    }
+                    if (chains_n >= chains_m) {
+                        chains_m = chains_m ? chains_m * 2 : 64;
+                        chains_arr = (mem_chain_t *)realloc(chains_arr,
```

Cppcheck (2.20.0)

[error] Common realloc mistake

(memleakOnRealloc)